django/db/models/query_utils.py

```diff
@@ -169,6 +173,94 @@ def _check_parent_chain(self, instance, name):
         return None
 
 
+class RegisterLookupMixin(object):
+    def _get_lookup(self, lookup_name):
+        try:
+            return self.class_lookups[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class' class_lookups.
+            for parent in inspect.getmro(self.__class__):
+                if 'class_lookups' not in parent.__dict__:
+                    continue
+                if lookup_name in parent.class_lookups:
+                    return parent.class_lookups[lookup_name]
+        except AttributeError:
+            # This class didn't have any class_lookups
+            pass
+        return None
+
+    def _get_transform(self, lookup_name):
+        # TODO(Josh): `_get_transform` and `_get_lookup` need to be combined somehow
+        try:
+            return self.class_transforms[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class' class_lookups.
+            for parent in inspect.getmro(self.__class__):
+                if 'class_transforms' not in parent.__dict__:
+                    continue
+                if lookup_name in parent.class_transforms:
+                    return parent.class_transforms[lookup_name]
+        except AttributeError:
+            # This class didn't have any class_transforms
+            pass
+        return None
+
+    def get_lookup(self, lookup_name):
+        from django.db.models.lookups import Lookup
```

Comment:
I think it is a mistake Lookup is a subclass of RegisterLookupMixin. Because registering lookups or transforms on a Lookup doesn't make any sense.